ui/src/UserAnnotation.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import Annotation from "./Annotation";
+import { ScholarReaderContext } from "./state";
+import { Annotation as AnnotationObject } from "./types/api";
+import UserAnnotationTooltipBody from "./UserAnnotationTooltipBody";
+
+interface UserAnnotationProps {
+  annotation: AnnotationObject;
+}
+
+export class UserAnnotation extends React.Component<UserAnnotationProps> {
+  static contextType = ScholarReaderContext;
+  context!: React.ContextType<typeof ScholarReaderContext>;
+
+  onKeyDown(e: React.KeyboardEvent<HTMLDivElement>) {
+    const { deleteAnnotation, updateAnnotation } = this.context;
+    if (e.key !== undefined) {
+      if (e.key === "Backspace" || e.key === "Delete") {
+        deleteAnnotation(this.props.annotation.id);
+      } else if (e.key === "c") {
+        updateAnnotation(this.props.annotation.id, {
+          ...this.props.annotation,
+          type: "citation"
+        });
+      } else if (e.key === "s") {
+        updateAnnotation(this.props.annotation.id, {
+          ...this.props.annotation,
+          type: "symbol"
+        });
+      }
+    }
+  }
+
+  onDeselect() {
+    const { setSelectedAnnotation, selectedAnnotation } = this.context;
+    if (
+      selectedAnnotation !== null &&
+      selectedAnnotation.id === this.props.annotation.id
+    ) {
+      setSelectedAnnotation(null);
+    }
+  }
+
+  render() {
+    let typeClass = "";
+    if (this.props.annotation.type === "citation") {
+      typeClass = "citation-user-annotation";
+    } else if (this.props.annotation.type === "symbol") {
+      typeClass = "symbol-user-annotation";
+    }
```

Comment:
I like using the [classnames](https://www.npmjs.com/package/classnames) package for this.  It'd allow you do do something like:

```TypeScript
const className = classNames("user-annottation", { 
    "citation-user-annotation": this.props.annotation.type === "citation",
    "symbol-user-annotation": this.props.annotation.type === "symbol"
});
```

I don't think you need to change this now though.  Just a potential `TODO` for down the road, particularly if you find yourself doing this kind of thing often.

Reply:
Nice! I'll integrate this. The existing code was feeling a little clunky.